### ISSUE TITLE
Regular expression denial of service vulnerability found in glob-parent 

### ISSUE THREAD

**Issue by Ibabalola** [nicolo-ribaudo/chokidar-2#5]
```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Regular expression denial of service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ glob-parent                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @babel/cli [dev]                                             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @babel/cli > @nicolo-ribaudo/chokidar-2 > glob-parent        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1751                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Overview**
glob-parent before 5.1.2 has a regular expression denial of service vulnerability. The enclosure regex used to check for strings ending in enclosure containing path separator.

**Remediation**
Upgrade to version 5.1.2 or later

**Comment by nicolo-ribaudo** [comment]
I _assume_ you are using a Node.js version higher than 6. In that case the CVE doesn't affect you and you can safely ignore the warning: in Node.js >= 8, Babel doesn't use this package but uses `chokidar@3` directly.

If you are still using Node.js 6, I highly recommend updating to a more recent Node.js version.

**Comment by ceisele-r** [comment]
@nicolo-ribaudo is there still anything that could be done? Otherwise, this would mean that everybody using @babel/cli now always gets a high vulnerability warning.

According to the changelog, it seems glob-parent v5 (https://github.com/gulpjs/glob-parent/releases/tag/v5.0.0) still supports Node 6 (as it says it dropped support for `Node < 6`).
Wouldn't it therefore be possible to update `glob-parent` to `v5.1.2` at least and by that, fix the issue? Or does this package needs to support Node versions lower than 6?

**Comment by nicolo-ribaudo** [comment]
I can carefully look into upgrading to v5.1.2.

However, this package isn't currently affected by the vulnerability: the fact that security tools report it is a bug in those security tools, which assume that the unsafe code is always used. Ideally they should let specify "if package X is only used by package Y, then X's vulnerability doesn't apply".

We are lucky here that 5.1.2 still supports Node.js 6 so I can try updating it, but you might consider reporting this as a feature request to the security check tools you are using.

**Comment by ceisele-r** [comment]
Thanks for looking into the upgrade.

I understand your point but the tool reporting this is just `npm audit` (that also implicitly runs with an `npm install`). So nothing special used there.

**Comment by stefan-squareweave** [comment]
> "if package X is only used by package Y, then X's vulnerability doesn't apply".

It turns out one of my colleagues has built an audit package to do this: https://www.npmjs.com/package/audit-app. I'm ignoring this warning using a .auditapprc.json file like this:

```json
{
  "packageManager": "npm",
  "ignore": ["1751|@babel/cli>@nicolo-ribaudo/chokidar-2>glob-parent"]
}

```

**Comment by nicolo-ribaudo** [comment]
Oh awesome!

Btw, we just released `@babel/cli@7.14.5` which should make this warning disappear.

**Comment by Ibabalola** [comment]
Thanks @nicolo-ribaudo I can confirm that `npm install @babel/cli@7.14.5` has removed the CVE warning from the `npm audit` within my project
